checkfiles.py

```diff
@@ -1078,21 +1081,24 @@ def run(out, err, url, username, password, encValData, mirror, search_query, fil
         jobs = fetch_files(session, url, search_query, out, include_unexpired_upload, file_list, local_file)
         if not json_out:
             headers = '\t'.join(['Accession', 'Lab', 'Errors', 'Aliases', 'Upload URL',
-                                'Upload Expiration'])
+                                 'Upload Expiration'])
             out.write(headers + '\n')
             out.flush()
         for job in imap(functools.partial(check_file, config, session, url), jobs):
             if not dry_run:
                 patch_file(session, url, job)
-
-            if not job.get('skip'):
-                errors_string = str(job.get('errors', {'errors': None}))
-            else:
-                errors_string = str({'errors': 'status have not been changed, the file check was skipped due to the file unavailability on S3'})
+            errors_string = {'errors': None}
```

Comment:
Eliminating the duplicated str functions was a good idea but the name should be errors_dict.  It technically doesn't get turned into a string until the report. +)

Also we didn't need the job.get('errors') because we know that all jobs have the errors key.  It happens when they are built in the fetch_files function.  That's a dividing line between using dict.get or not.  If we don't know a key exists but we want to handle the failure gracefully then use get.  In this case we add it specifically, so it might be better to have the code break if 'errors' gets deleted.  If a default value get deleted we probably want to know about it.

Reply:
Fixed